pg-index-health/src/test/java/io/github/mfvanek/pg/checks/host/NotValidConstraintsOnHostTest.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2019-2024. Ivan Vakhrushev and others.
+ * https://github.com/mfvanek/pg-index-health
+ *
+ * This file is a part of "pg-index-health" - a Java library for
+ * analyzing and maintaining indexes health in PostgreSQL databases.
+ *
+ * Licensed under the Apache License 2.0
+ */
+
+package io.github.mfvanek.pg.checks.host;
+
+import io.github.mfvanek.pg.common.maintenance.DatabaseCheckOnHost;
+import io.github.mfvanek.pg.common.maintenance.Diagnostic;
+import io.github.mfvanek.pg.model.PgContext;
+import io.github.mfvanek.pg.model.constraint.Constraint;
+import io.github.mfvanek.pg.model.constraint.ConstraintType;
+import io.github.mfvanek.pg.support.DatabaseAwareTestBase;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static io.github.mfvanek.pg.support.AbstractCheckOnHostAssert.assertThat;
+
+public class NotValidConstraintsOnHostTest extends DatabaseAwareTestBase {
+
+    private final DatabaseCheckOnHost<Constraint> check = new NotValidConstraintsOnHost(getPgConnection());
+
+    @Test
+    void shouldSatisfyContract() {
+        assertThat(check)
+                .hasType(Constraint.class)
+                .hasDiagnostic(Diagnostic.NOT_VALID_CONSTRAINTS)
+                .hasHost(getHost());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = {PgContext.DEFAULT_SCHEMA_NAME, "custom"})
+    void onDatabaseWithThem(final String schemaName) {
+        executeTestOnDatabase(schemaName, dbp -> dbp.withNotValidConstraints().withUniqueConstraintOnSerialColumn(), ctx ->
+                assertThat(check)
+                        .executing(ctx)
+                        .hasSize(2)
+                        .containsExactly(
+                                Constraint.of(ctx.enrichWithSchema("accounts"), "c_accounts_chk_client_id_not_validated_yet",
+                                        ConstraintType.CHECK),
+                                Constraint.of(ctx.enrichWithSchema("accounts"), "c_accounts_fk_client_id_not_validated_yet",
+                                        ConstraintType.FOREIGN_KEY)
+                        )
```

Comment:
Add `validate constraint` call and after that recheck again (should be empty).  
See https://github.com/mfvanek/pg-index-health/pull/374#issuecomment-2081961766

For example
```sql
ALTER TABLE distributors VALIDATE CONSTRAINT distfk;
```
https://www.postgresql.org/docs/16/sql-altertable.html

Reply:
```java

//example..
        statement.execute(String.format("alter table %1$s.accounts validate constraint c_accounts_fk_client_id_not_validated_yet;",
                schemaName));
        statement.execute(String.format("alter table %1$s.accounts validate constraint c_accounts_chk_client_id_not_validated_yet;",
                schemaName));
```

I think it's okay to create a new statement that calls the valid constraint and test it. What are your thought?

Reply:
> I think it's okay to create a new statement that calls the valid constraint and test it. What are your thoughts?

I'd use option with `ExecuteUtils.executeOnDatabase(getDataSource(), statement -> statement.execute("vacuum analyze"));`
https://github.com/mfvanek/pg-index-health/blob/master/pg-index-health/src/test/java/io/github/mfvanek/pg/support/ExecuteUtils.java#L29

Reply:
```java
    @ParameterizedTest
    @ValueSource(strings = {PgContext.DEFAULT_SCHEMA_NAME, "custom"})
    void onDatabaseWithThem(final String schemaName) {
        executeTestOnDatabase(schemaName, dbp -> dbp.withNotValidConstraints().withUniqueConstraintOnSerialColumn(), ctx -> {
            assertThat(check)
                    .executing(ctx)
                    .hasSize(2)
                    .containsExactly(
                            Constraint.of(ctx.enrichWithSchema("accounts"), "c_accounts_chk_client_id_not_validated_yet", ConstraintType.CHECK),
                            Constraint.of(ctx.enrichWithSchema("accounts"), "c_accounts_fk_client_id_not_validated_yet", ConstraintType.FOREIGN_KEY)
                    );

            ExecuteUtils.executeOnDatabase(DatabaseAwareTestBase.getDataSource(), statement -> {
                statement.execute(String.format("alter table %1$s.accounts validate constraint c_accounts_fk_client_id_not_validated_yet;", schemaName));
                statement.execute(String.format("alter table %1$s.accounts validate constraint c_accounts_chk_client_id_not_validated_yet;", schemaName));
            });

            assertThat(check)
                    .executing(ctx)
                    .isEmpty();
        });
    }
```

proceed like this?

Reply:
> proceed like this?

Yeah!